src/conf_mode/interfaces-bridge.py

```diff
@@ -166,31 +131,37 @@ def verify(bridge):
 
             if 'has_address' in interface_config:
                 raise ConfigError(error_msg + 'it has an address assigned!')
-            
-            if 'has_vlan' in interface_config:
-                raise ConfigError(error_msg + 'it has an VLAN subinterface assigned!')
-            
-            # VLAN-aware bridge members must not have VLAN interface configuration
-            if 'native_vlan' in interface_config:
-                vlan_aware = True
-            
-            if 'allowed_vlan' in interface_config:
-                vlan_aware = True
-            
-            if vlan_aware and 'wlan' in interface:
-                raise ConfigError(error_msg + 'VLAN aware cannot be set!')
-            
-            if 'allowed_vlan' in interface_config:
-                for vlan in interface_config['allowed_vlan']:
-                    if re.search('[0-9]{1,4}-[0-9]{1,4}', vlan):
-                        vlan_range = vlan.split('-')
-                        if int(vlan_range[0]) <1 and int(vlan_range[0])>4094:
-                            raise ConfigError('VLAN ID must be between 1 and 4094')
-                        if int(vlan_range[1]) <1 and int(vlan_range[1])>4094:
-                            raise ConfigError('VLAN ID must be between 1 and 4094')
-                    else:
-                        if int(vlan) <1 and int(vlan)>4094:
-                            raise ConfigError('VLAN ID must be between 1 and 4094')
+
+            if 'enable_vlan' in bridge:
+                if 'has_vlan' in interface_config:
+                    raise ConfigError(error_msg + 'it has an VLAN subinterface assigned!')
+
+                if 'wlan' in interface:
+                    raise ConfigError(error_msg + 'VLAN aware cannot be set!')
+
+                if 'allowed_vlan' in interface_config:
+                    for vlan in interface_config['allowed_vlan']:
+                        if re.search('[0-9]{1,4}-[0-9]{1,4}', vlan):
+                            vlan_range = vlan.split('-')
+                            if int(vlan_range[0]) <1 and int(vlan_range[0])>4094:
+                                raise ConfigError('VLAN ID must be between 1 and 4094')
+                            if int(vlan_range[1]) <1 and int(vlan_range[1])>4094:
+                                raise ConfigError('VLAN ID must be between 1 and 4094')
+                        else:
+                            if int(vlan) <1 and int(vlan)>4094:
+                                raise ConfigError('VLAN ID must be between 1 and 4094')
+            else:
+                if 'allowed_vlan' in interface_config:
```

Comment:
Please use:
```
for option in ['allowed_vlan', 'native_vlan']:
    if option in interface_config:
        raise ConfigError('Can not use VLAN options on non VLAN aware bridge')
```

That eases any type of maintenance...

Reply:
done